DataStructures.Tests/AATreeTests.cs

```diff
@@ -0,0 +1,221 @@
+
+            Assert.IsTrue(tree.Contains(6));
+            Assert.IsFalse(tree.Contains(999));
+        }
+
+        [Test]
+        public static void Contains_EmptyTree_ReturnsFalse()
+        {
+            var tree = new AATree<int>();
+
+            Assert.IsFalse(tree.Contains(999));
+        }
+
+        [Test]
+        public static void GetMax_NonEmptyTree_ReturnsCorrectAnswer()
+        {
+            var tree = new AATree<int>();
+            tree.AddRange(new int[] { 1, 2, 3, 4, 5, 6, 7, 8, 9, 10 });
+
+            Assert.AreEqual(10, tree.GetMax());
+        }
+
+        [Test]
+        public static void GetMax_EmptyTree_ThrowsCorrectException()
+        {
+            var tree = new AATree<int>();
+            Assert.Throws<InvalidOperationException>(() => tree.GetMax());
+        }
+
+        [Test]
+        public static void GetMin_NonEmptyTree_ReturnsCorrectAnswer()
+        {
+            var tree = new AATree<int>();
+            tree.AddRange(new int[] { 1, 2, 3, 4, 5, 6, 7, 8, 9, 10 });
+
+            Assert.AreEqual(1, tree.GetMin());
+        }
+
+        [Test]
+        public static void GetMin_EmptyTree_ThrowsCorrectException()
+        {
+            var tree = new AATree<int>();
+            Assert.Throws<InvalidOperationException>(() => tree.GetMin());
+        }
+
+        [Test]
+        public static void GetKeysInOrder_NonEmptyTree_ReturnsCorrectAnswer()
+        {
+            var tree = new AATree<int>();
+            tree.AddRange(new int[] { 1, 2, 3, 4, 5, 6, 7, 8, 9, 10 });
+
+            var expected = new int[] { 1, 2, 3, 4, 5, 6, 7, 8, 9, 10 };
+            var actual = tree.GetKeysInOrder();
+            Assert.IsTrue(Enumerable.SequenceEqual(expected, actual));
+        }
+
+        [Test]
+        public static void GetKeysInOrder_EmptyTree_ReturnsCorrectAnswer()
+        {
+            var tree = new AATree<int>();
+            Assert.IsTrue(tree.GetKeysInOrder().ToList().Count == 0);
+        }
+
+        [Test]
+        public static void GetKeysPreOrder_NonEmptyTree_ReturnsCorrectAnswer()
+        {
+            var tree = new AATree<int>();
+            tree.AddRange(new int[] { 1, 2, 3, 4, 5, 6, 7, 8, 9, 10 });
+
+            var expected = new int[] { 4, 2, 1, 3, 6, 5, 8, 7, 9, 10 };
+            var actual = tree.GetKeysPreOrder();
+            Assert.IsTrue(Enumerable.SequenceEqual(expected, actual));
+        }
+
+        [Test]
+        public static void GetKeysPreOrder_EmptyTree_ReturnsCorrectAnswer()
+        {
+            var tree = new AATree<int>();
+            Assert.IsTrue(tree.GetKeysPreOrder().ToList().Count == 0);
+        }
+
+        [Test]
+        public static void GetKeysPostOrder_NonEmptyTree_ReturnsCorrectAnswer()
+        {
+            var tree = new AATree<int>();
+            tree.AddRange(new int[] { 1, 2, 3, 4, 5, 6, 7, 8, 9, 10 });
+
+            var expected = new int[] { 1, 3, 2, 5, 7, 10, 9, 8, 6, 4 };
```

Comment:
Codacy found an issue: [Remove the array type; it is redundant.](https://app.codacy.com/manual/siriak/C-Sharp/pullRequest?prid=6367315)